brainclouds2s.js

```diff
@@ -29,60 +27,47 @@ function s2sRequest(context, json, callback)
         }
     }
 
-    var req = https.request(options, res =>
-    {
+    var req = https.request(options, res => {
         var data = ''
-        
+
         // A chunk of data has been recieved.
-        res.on('data', chunk =>
-        {
+        res.on('data', chunk => {
             data += chunk
         })
-        
+
         // The whole response has been received. Print out the result.
-        res.on('end', () =>
-        {
-            if (context.logEnabled)
-            {
+        res.on('end', () => {
+            if (context.logEnabled) {
                 console.log(`[S2S RECV ${context.appId}] ${data}`)
             }
-            if (callback)
-            {
-                if (data)
-                {
-                    
+            if (callback) {
+                if (data) {
+
                     try {
                         let dataJson = JSON.parse(data)
-                        callback(context, dataJson)
-                    } 
-                    catch (error) 
-                    {
-                        if (context.logEnabled)
-                        {
+                        callCallback(callback, context, dataJson)
+                    }
+                    catch (error) {
+                        if (context.logEnabled) {
                             console.log(`[S2S Error ${context.appId}] ${error}`)
                         }
-                        if (callback)
-                        {
-                            callback(context, null)
+                        if (callback) {
```

Comment:
You can remove all these `if (callback)` checks IF the only purpose is to know whether or not to call `callCallback()`... just call `callCallback()` and if the callback is null, it'll be a no-op.